### ISSUE TITLE
ENH refactor the way normalization is being applied to the datasets

### ISSUE THREAD

**Issue by zaccharieramzi** [benchopt/benchmark_resnet_classif#24]
Currently, there is a lot of copy-pasting that was introduced by https://github.com/benchopt/benchmark_resnet_classif/pull/19 when it comes to handling the normalization.

Basically, we want to have the same normalization for all datasets, but not apply it at the same times.
In particular, we want to be able to apply it after data augmentation in the case of the training set when fitting the model.

**Comment by tomMoral** [comment]
yeah I agree that would be nice to refactor, right now we are applying the normalisation in three different places.
